### DISCUSSION TITLE
Is there a way to use pytorch with MEEP

### DISCUSSION THREAD

**Discussion by 10UAikawa** [NanoComp/meep#2653] [Q&A]
So, I have MEEP working on my windows laptop using WSL. I have MEEP working and also pytorch working. The problem I am running into is not being able to use them together. I think when installing MEEP it requires to download it with miniconda and puts it in its own virtual environment. I just want to know how I can add pytorch to that environment.

**Comment by zekishaw** [comment]
Im guessing its been tried but have you conda installed/pip installed pytorch in the environment with MEEP? 

E.g 

If you used conda create -n meep, 

Then 

Conda activate meep

Conda install pytorch


pip install pytorch [if conda doesn't work]

**Reply by zekishaw** [reply]
Also you need to be in WSL to activate MEEP and so will need to install pytorch on the conda environment inside WSL

**Reply by 10UAikawa** [reply]
I have Pytorch installed on my WSL, I think it might be a problem of including it into the MEEP environment. When MEEP is installed its put in its own virtual environment, so I just don't know how to include Pytorch into MEEPs virtual environment.

---------------------------------This is from the MEEP website--------------------------------------------------------
Next, we create a Conda environment for PyMeep (serial version) to isolate it from other Python libraries that may be installed.

Then this command in run: 
conda create -n mp -c conda-forge pymeep pymeep-extras

This creates an environment called "mp" (you can name this anything you like) with PyMeep and all its dependencies. This will default to the version of Python in your Miniconda installation.
--------------------------------------------------------------------------------------------------------------------------

So, I just want to include Pytorch into the mp environment.

**Reply by smartalecH** [reply]
As already mentioned, you need to install both in the same environment. Maybe read over the conda docs to review the difference between creating environments and managing environments.
